src/collector/top_score_collector.rs

```diff
@@ -86,12 +85,15 @@ where
 /// The `TopDocs` collector keeps track of the top `K` documents
 /// sorted by their score.
 ///
-/// The implementation is based on a `BinaryHeap`.
-/// The theoretical complexity for collecting the top `K` out of `n` documents
-/// is `O(n log K)`.
+/// The implementation is based on a repeatedly truncating on the median after K * 2 documents
```

Comment:
So this basically trades having to store `2K` instead of `K` documents against less computational sorting work?

Reply:
`2K`, so we can drop elements when truncating the median